balancer/pickfirst/pickfirstleaf/pickfirstleaf_ext_test.go

```diff
@@ -66,20 +66,57 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
+// testServer is a server than can be stopped and resumed without closing
+// the listener. This guarantees the same port number (and address) is used
+// after restart. When a server is stopped, it accepts and closes all tcp
+// connections from clients.
+type testServer struct {
+	stubserver.StubServer
+	lis *testutils.RestartableListener
+}
+
+func (s *testServer) stop() {
+	s.lis.Stop()
+}
+
+func (s *testServer) resume() {
+	s.lis.Restart()
+}
+
+func newTestServer() (*testServer, error) {
+	l, err := testutils.LocalTCPListener()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We *could* accept a `testing.T` here, make this a test helper and call `t.Fatalf` here if creating the listener fails. That would simplify callsites since they wont have to handle two return values.